src/components/SecretsTable/SecretsTable.js

```diff
@@ -28,124 +39,160 @@ const {
 } = DataTable;
 
 const SecretsTable = props => {
-  const { handleDelete, handleNew, loading, secrets } = props;
+  const {
+    handleDelete,
+    handleNew,
+    loading,
+    secrets,
+    selectedNamespace
+  } = props;
 
-  const initialHeaders = [
-    { key: 'secret', header: 'Secret' },
-    { key: 'annotations', header: 'Annotations' },
-    { key: 'add', header: <Add /> }
-  ];
+  const initialHeaders =
+    selectedNamespace !== ALL_NAMESPACES
+      ? [
+          { key: 'secret', header: 'Secret' },
+          { key: 'annotations', header: 'Annotations' }
+        ]
+      : [
+          { key: 'secret', header: 'Secret' },
+          { key: 'namespace', header: 'Namespace' },
+          { key: 'annotations', header: 'Annotations' }
+        ];
 
   const secretsFormatted = secrets.map(secret => {
     let annotations = '';
     Object.keys(secret.annotations).forEach(function annotationSetup(key) {
       annotations += `${key}: ${secret.annotations[key]}\n`;
     });
-    return {
-      id: `${secret.namespace}:${secret.name}`,
-      secret: secret.name,
-      annotations,
-      add: <Close />
-    };
+    return selectedNamespace !== ALL_NAMESPACES
```

Comment:
try to avoid duplication, especially in data formatting, as any update is increasingly likely to cause bugs the more places the code needs to be changed to match

```js
const formattedSecret = {
  id: `${secret.namespace}:${secret.name}`,
  secret: secret.name,
  annotations
};

if (selectedNamespace === ALL_NAMESPACES) {
   formattedSecret.namespace = secret.namespace;
}

return formattedSecret;
```